### PR TITLE
[Help Editor] Fix Help Editor module preview & help section rendering

### PR DESCRIPTION
**Pull request by zaliqarosli**
This pull request lets help section for all Loris pages and preview feature of Help Editor module to render Markdown content. It uses the Markdown and DisplayHelp React components to display and render the form content.

This PR lets modules take help content from the markdown files in their help directories, while instruments' help content is managed on the front-end, using the Help Editor module. This is up for discussion.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -49,7 +51,8 @@ var config = [{
     './modules/help_editor/js/columnFormatter.js': './modules/help_editor/jsx/columnFormatter.js',
     './modules/brainbrowser/js/Brainbrowser.js': './modules/brainbrowser/jsx/Brainbrowser.js',
     './modules/data_integrity_flag/js/index.js': './modules/data_integrity_flag/jsx/index.js',
-    './modules/imaging_uploader/js/index.js': './modules/imaging_uploader/jsx/index.js'
```
Comment (kongtiaowang) [inline webpack.config.js]: Why you removed this from here?
'./modules/imaging_uploader/js/index.js'

Comment (zaliqarosli) [reply webpack.config.js]: it got moved to line 54 below

File: modules/help_editor/jsx/help_editor_helper.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class EditContent extends React.Component {
```
Comment (zaliqarosli) [inline modules/help_editor/jsx/help_editor_helper.js]: Thought: not standard to have React component and jQuery in one file under jsx. For the next iteration, would like to pull this React component out into its own file as an exported component in modules/help_editor/jsx and put help_editor_helper.js back in modules/help_editor/js with no jsx.

Comment (zaliqarosli) [inline modules/help_editor/jsx/help_editor_helper.js]: would also like to change class name to PreviewHelp

File: htdocs/js/helpHandler.js [hunk 3]
```diff
@@ -57,7 +46,7 @@ $(document).ready(function() {
       div.appendChild(wrap);
       div.appendChild(btn);
 
-      // Markdown format content came from the filesystem and can't
+      // Markdown format content came from the filesystem and can
```
Comment (driusan) [inline htdocs/js/helpHandler.js]: I don't think this comment is relevent anymore now that markdown content can come from either the filesystem or database. You can just remove it rather than modify it.

File: modules/help_editor/jsx/help_editor_helper.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          <hr />
+          <p>Last updated: {date.getFullYear()}-{date.getMonth() + 1}-{date.getDate()} {date.getHours()}:{date.getMinutes()}:{date.getSeconds()}</p>
+	</pre>
+	<button className="btn btn-default" id="helpclose" onClick={removeDiv}>Close</button>
+      </div>
+    );
+  }
+
+}
+$(document).ready(function() {
```
Comment (driusan) [inline modules/help_editor/jsx/help_editor_helper.js]: As @ZainVirani said, it's weird to have this mixture of jquery and React in one file. Why was this moved out of the other file?

Comment (ZainVirani) [reply modules/help_editor/jsx/help_editor_helper.js]: I said this??

Comment (driusan) [reply modules/help_editor/jsx/help_editor_helper.js]: Oops.. I guess my eyes didn't read past "za". I was referring to the comment on line 1.

File: jsx/DisplayHelp.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        <div>
+          <hr />
+          <p>Last updated: {date}</p>
+        </div>
+      );
+    }
+    return (
+      <div>
+	<h3>{this.props.Topic ? this.props.Topic : ''}</h3>
+	<Markdown content={this.props.Content} />
```
Comment (driusan) [inline jsx/DisplayHelp.js]: This should be indented to 1 level further than the div.

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,4015 @@
+{
```
Comment (driusan) [inline package-lock.json]: This file shouldn't be changed

**Comment by zaliqarosli** [comment]
@driusan I made the changes, please review when you get a chance!

**Comment by zaliqarosli** [comment]
this PR needs to grab changes from https://github.com/aces/Loris/pull/3345 . waiting for minor to be updated with 19.0-dev

File: modules/help_editor/js/help_editor_helper.js [hunk 7]
```diff
@@ -1,39 +1,19 @@
...
-        $(div).remove();
-        e.preventDefault();
-        }) ;
-     e.preventDefault();
-
-});
+    document.getElementById('page').appendChild(div);
+    div.setAttribute("class", "help-content");
+    div.setAttribute("id", "preview");
+    ReactDOM.render(RPreviewHelp({Title: title, Content: content, MyDate: myDate}), document.getElementById("preview"));
```
Comment (ZainVirani) [inline modules/help_editor/js/help_editor_helper.js]: Should this be PreviewHelp?

**Comment by ridz1208** [comment]
@zaliqarosli whats the ETA on this ?

**Comment by zaliqarosli** [comment]
@ridz1208 in this PR, the help feature of all modules get content for the markdown files in their help directory while all instruments get content from the DB help table. this means that users can only edit help content for instruments on the front-end. the discussion would be whether it might be useful to also allow help content for modules to be editable from the front-end

**Comment by zaliqarosli** [comment]
closing PR and making another one
